### PR TITLE
Pomdp almsure fix

### PR DESCRIPTION
**Pull request by sjunges**
Fix of a fix i made last spring :-)

### CODE REVIEW

File: src/storm-pomdp/analysis/IterativePolicySearch.cpp [hunk 1]
```diff
@@ -238,34 +234,36 @@ namespace storm {
...
-                    subexprreachSwitch.push_back(followVarExpressions[pomdp.getObservation(state)]);
                     subexprreachNoSwitch.push_back(followVarExpressions[pomdp.getObservation(state)]);
+
                     for (auto const &entries : pomdp.getTransitionMatrix().getRow(rowindex)) {
-                        if (pomdp.getObservation(entries.getColumn()) == pomdp.getObservation(state)) {
+                        if (!delayedSwitching || pomdp.getObservation(entries.getColumn()) != pomdp.getObservation(state)) {
                             subexprreachSwitch.push_back(continuationVarExpressions.at(entries.getColumn()));
                         } else {
                             subexprreachSwitch.push_back(reachVarExpressions.at(entries.getColumn()));
+                            subexprreachSwitch.push_back(continuationVarExpressions.at(entries.getColumn()));
```
Comment (tquatmann) [inline src/storm-pomdp/analysis/IterativePolicySearch.cpp:253]: Is it intended that this line is the same as in the if-branch?

File: src/storm-pomdp/analysis/IterativePolicySearch.cpp [hunk 2]
```diff
@@ -85,7 +85,7 @@ namespace storm {
         template <typename ValueType>
         bool IterativePolicySearch<ValueType>::initialize(uint64_t k) {
             STORM_LOG_INFO("Start intializing solver...");
-
+            bool delayedSwitching = false; // Notice that delayed switching is currently broken.
```
Comment (tquatmann) [inline src/storm-pomdp/analysis/IterativePolicySearch.cpp:88]: Maybe elaborate what is broken?

**Comment by tquatmann** [comment]
Changes look good to me; except for the two minor comments and the failing compilation 😅

File: src/storm-pomdp/analysis/IterativePolicySearch.cpp (hunk 2 above)
Comment (sjunges) [reply src/storm-pomdp/analysis/IterativePolicySearch.cpp]: Done (I dont really know what is broken. I know what I want to do first in order to fix this ;-))

File: src/storm-pomdp/analysis/IterativePolicySearch.cpp (hunk 1 above)
Comment (sjunges) [reply src/storm-pomdp/analysis/IterativePolicySearch.cpp:252]: Yes, added a comment.

**Pull request merged**